aiidalab/app.py

```diff
@@ -0,0 +1,647 @@
+# -*- coding: utf-8 -*-
+"""Module to manage AiiDA lab apps."""
+
+import re
+import os
+from os import path
+from time import sleep
+from collections import OrderedDict
+
+import requests
+import ipywidgets as ipw
+from dulwich.repo import Repo
+
+
+class AppNotInstalledException(Exception):
+    pass
+
+
+class Version(ipw.VBox):
+    """Class to choose app's version."""
+
+    def __init__(self):
+        self.change_btn = ipw.Button(description="choose seleted")
+        self.selected = ipw.Select(
+            options={},
+            description='Select version',
+            disabled=False,
+            style={'description_width': 'initial'},
+        )
+        self.info = ipw.HTML('')
+        super().__init__([self.selected, self.change_btn, self.info])
+
+
+class AiidalabApp():  # pylint: disable=attribute-defined-outside-init,too-many-public-methods,too-many-instance-attributes
+    """Class to manage AiiDA lab app."""
+
+    def __init__(self, name, app_data, aiidalab_apps):  #, custom_update=False):
```

Comment:
In my proposed revisions, the app will be identified by its path and nothing else. I assume we are going to merge this regardless and then I will apply the breaking change?

Reply:
I would say so.